gitbook-docs/connection.md

```diff
@@ -1,63 +1,76 @@
 # Discovery and Connection Establishment
 
-#### Service Discovery
+## Service Discovery
 
-A Signal K server SHOULD advertise its services over mDNS/Bonjour. The server MUST use the service types
+A Signal K server SHOULD advertise its services using DNS Service Discovery (DNS-SD) via Mutlicast DNS (mDNS); also
+known as Bonjour. The server MUST provide DNS [Service (SRV) Records](https://en.wikipedia.org/wiki/SRV_record) and
+[Text (TXT) Records](https://en.wikipedia.org/wiki/TXT_record) describing the Signal K interfaces it provides. These
+service identifiers are:
 
-* `_signalk-http._tcp` for http API
-* `_signalk-ws._tcp` for WebSocket
-* `_signalk-https._tcp` for HTTPS API
-* `_signalk-wss._tcp` for secure WebSocket
+* `_http._tcp` and/or `_https._tcp` for the server's web interface
+* `_signalk-http._tcp` and/or `_signalk-https._tcp` for the Signal K REST API
+* `_signalk-ws._tcp` and/or `_signalk-wss._tcp` for the WebSocket data stream
 
-Furthermore a server SHOULD advertise its web interface with normal Bonjour convention `_http._tcp` and `_https._tcp`.
+One of the following parameters MUST be set, both MAY be set:
 
-A sample Bonjour record output, dumped using avahi-discover:
+* `vessel_uuid` is the Signal K UUID which uniquely identifies the vessel
+* `vessel_mmsi` is the MMSI assigned to the vessel
+
+A vessel may have one or both of these values. One or both MUST be available when advertising the Signal K server.
+
+The following parameters SHOULD be set in the TXT record:
+
+* `server` is the name of the Signal K server software, e.g. signalk-server-node
+* `version` is the version of the Signal K server software
+
+These parameters identify the server and its version. They are purely informational.
+
+The following parameters MAY be set in the TXT record:
+
+* `vessel_name` is the registered name of the vessel
+* `vessel_brand` is the name of the manufacturer of the vessel
+* `vessel_type` is the model of the vessel
+
+An example DNS-SD record set might look something like this
 
 ```
-Service data for service 'signalk-http (2)' of type '_signalk-http._tcp' in domain 'local' on 4.0:
+Service data for service 'signalk-http' of type '_signalk-http._tcp' in domain 'local' on 4.0:
     Host 10-1-1-40.local (10.1.1.40),
-    port 8080,
+    port 80,
     TXT data: [
+        'vessel_name=Volare',
+        'vessel_brand=Friendship',
+        'vessel_type=22',
+        'vessel_mmsi=503123456',
         'vessel_uuid=urn:mrn:signalk:uuid:6b0e776f-811a-4b35-980e-b93405371bc5',
-        'version=v1.0.0',
-        'vessel_name=urn:mrn:signalk:uuid:6b0e776f-811a-4b35-980e-b93405371bc5',
-        'vessel_mmsi=urn:mrn:signalk:uuid:6b0e776f-811a-4b35-980e-b93405371bc5',
-        'server=signalk-server',
-        'path=/signalk'
```

Comment:
The node server doesn't include `path=/signalk`, so I removed it from the example. Should the server be providing it?